### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Vasyl-Poremchuk**

### CODE REVIEW

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: db/models.py [hunk 1]
```diff
@@ -1 +1,26 @@
 from django.db import models
+
+
+class Race(models.Model):
```
[the remark this answers is not in the document]
Comment (Vasyl-Poremchuk) [reply db/models.py:4]: Thanks for your review)

**Review by LiudmylaKulinchenko — APPROVED** [review]
Good job!

File: db/models.py [hunk 2]
```diff
@@ -1 +1,32 @@
...
+    name = models.CharField(max_length=255, unique=True)
+    description = models.TextField(null=True)
+
+
+class Player(models.Model):
+    nickname = models.CharField(max_length=255, unique=True)
+    email = models.EmailField(max_length=255)
+    bio = models.CharField(max_length=255)
+    race = models.ForeignKey(Race, on_delete=models.CASCADE)
+    guild = models.ForeignKey(Guild, on_delete=models.CASCADE, null=True)
```
Comment (Y-Havryliv) [inline db/models.py:31]: player should not be deleted when the guild is deleted

File: main.py [hunk 3]
```diff
@@ -1,10 +1,52 @@
...
+            current_guild = None
+        else:
+            if not Guild.objects.filter(name=guild["name"]).exists():
+                Guild.objects.create(
+                    name=guild["name"],
+                    description=guild["description"],
+                )
+            current_guild = Guild.objects.get(name=guild["name"])
+
+        if not Player.objects.filter(nickname=player).exists():
```
Comment (Y-Havryliv) [inline main.py:42]: It' no need to check if a player already exists

File: db/models.py [hunk 4]
```diff
@@ -1 +1,32 @@
...
+    created_at = models.DateTimeField(auto_now=True)
```
Comment (Y-Havryliv) [inline db/models.py:32]: `auto_now_add` would be a better option here

**Review by Y-Havryliv — CHANGES_REQUESTED** [review]

File: db/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nattalli) [reply db/models.py:4]: @LiudmylaKulinchenko Btw, It is not required, solution without choices is correct as well

File: db/models.py [hunk 5]
```diff
@@ -1 +1,32 @@
...
+    name = models.CharField(max_length=255, unique=True)
+    description = models.TextField(null=True)
+
+
+class Player(models.Model):
+    nickname = models.CharField(max_length=255, unique=True)
+    email = models.EmailField(max_length=255)
+    bio = models.CharField(max_length=255)
+    race = models.ForeignKey(Race, on_delete=models.CASCADE)
+    guild = models.ForeignKey(Guild, on_delete=models.PROTECT, null=True)
```
Comment (Nattalli) [inline db/models.py:31]: `models.SET_NULL` would be better here

Comment (Nattalli) [inline db/models.py:31]: add `related_name`, it's a best practice 😸

File: main.py [hunk 6]
```diff
@@ -1,10 +1,51 @@
...
+        race = player_info["race"]
+        skills = race["skills"]
+        guild = player_info["guild"]
+
+        if not Race.objects.filter(name=race["name"]).exists():
+            Race.objects.create(
+                name=race["name"],
+                description=race["description"],
+            )
+        current_race = Race.objects.get(name=race["name"])
```
Comment (Nattalli) [inline main.py:22]: can you optimize it? At least, you can make `race = Race.objects.create()...`. 
Then, you don't have to make an additional request to the database to get that value

File: main.py [hunk 7]
```diff
@@ -1,10 +1,51 @@
...
+        if not guild:
+            current_guild = None
+        else:
+            if not Guild.objects.filter(name=guild["name"]).exists():
+                Guild.objects.create(
+                    name=guild["name"],
+                    description=guild["description"],
+                )
+            current_guild = Guild.objects.get(name=guild["name"])
+
```
Comment (Nattalli) [inline main.py:41]: the same here. you can optimize

**Review by Nattalli — CHANGES_REQUESTED** [review]

File: db/models.py (hunk 5 above)
Comment (Nattalli) [reply db/models.py:31]: Please do not add this. Yes, this is a best practice, but the tests are not ready for such code(
Sorry...

**Review by Nattalli — APPROVED** [review]
Well done!

**Pull request closed without merging**